spinnman/processes/get_machine_process.py

```diff
@@ -171,52 +185,259 @@ def get_machine_details(self, boot_x, boot_y, width, height,
+                        chip_info.x, chip_info.y,
+                        chip_info.ethernet_ip_address)
+                    chip_info.ethernet_ip_address = None
                 else:
-                    machine.add_chip(self._make_chip(chip_info, machine))
-
-        removed_chips = []
-        if len(bad_ethernets) > 0:
-            msg = ""
-            for chip_info, claim in bad_ethernets:
-                chip = self._make_chip(chip_info, machine)
-                local_x, local_y = machine.get_local_xy(chip)
-                ethernet = machine.get_chip_at(
-                    chip.nearest_ethernet_x, chip.nearest_ethernet_y)
-                msg += "x:{} y:{} on board:{} claims it has ethernet {} " \
-                       "".format(local_x, local_y, ethernet.ip_address, claim)
-                removed_chips.append((chip.x, chip.y))
-            logger.warning(msg)
-
-            if ignore_bad_ethernets:
-                # No chips actually removed
-                removed_chips = []
+                    logger.warning(
+                        "Not using chip {}:{} as it has an unexpected "
+                        "ip address: {}", chip_info.x, chip_info.y,
+                        chip_info.ethernet_ip_address)
+                    continue
+            else:
+                machine.add_chip(self._make_chip(chip_info, machine))
+
         machine.validate()
-        return machine_repair(machine, repair_machine, removed_chips)
+        return machine_repair(machine, repair_machine)
 
-    def get_chip_info(self):
-        """ Get the chip information for the machine.
+    # Stuff below here is purely for dealing with ignores
 
-        .. note::
-            :py:meth:`get_machine_details` must have been called first.
+    def _process_ignore_links(self, machine):
         """
-        return self._chip_info
+        Processes the collection of ignore links to remove then from chipinfo
+
+        Converts any local x, y ipaddress to global xy
+
+        Discards any ignores which are known to have no\
+        affect based on the already read chip_info
+
+        Also removes any inverse linka
+
+        Logs all actions except for ignores with unused ip addresses
+
+        :param machine: An empty machine to handle wraparrounds
+        """
+        if len(self._ignore_links) == 0:
+            return
+
+        discarded = set()
+        for ignore in self._ignore_links:
+            if len(ignore) == 3:
+                local_x, local_y, link = ignore
+                ip = None
+            else:
+                local_x, local_y, link, ip = ignore
+            global_xy = self._ignores_local_to_global(
+                local_x, local_y, ip, machine)
+            if global_xy is None:
+                continue
+            chip_info = self._chip_info[global_xy]
+            if link in chip_info.working_links:
+                chip_info.working_links.remove(link)
+                logger.info("On chip {} ignoring link:{}",
```

Comment:
again, do we want to tell the end user we're ignoring the link they told us to ignore......... bit obsolete

Reply:
id put it as a debug instead of info/warning if you really want this

Reply:
leaving as 1 only people who use ignores are testers and then info is useful.

Reply:
i beg to differ that the only users are testers. end user who distrust a chip would use it. if it was just testers, we wouldn't have it in the default cfg.